urlbd77b745af_KokaKiwi_MCLauncher_tgz-pJ8-net_contrapunctus_lzma_BenchmarkJ8/src/SevenZip/LzmaBench.java

```diff
@@ -0,0 +1,480 @@
+        {
+            if (inSize >= ApprovedStart && InSize == 0)
+            {
+                Time = System.currentTimeMillis();
+                InSize = inSize;
+            }
+        }
+    }
+    
+    static final int kSubBits = 8;
+    
+    static int GetLogSize(int size)
+    {
+        for (int i = kSubBits; i < 32; i++)
+        {
+            for (int j = 0; j < 1 << kSubBits; j++)
+            {
+                if (size <= (1 << i) + (j << i - kSubBits))
+                {
+                    return (i << kSubBits) + j;
+                }
+            }
+        }
+        return 32 << kSubBits;
+    }
+    
+    static long MyMultDiv64(long value, long elapsedTime)
+    {
+        long freq = 1000; // ms
+        long elTime = elapsedTime;
+        while (freq > 1000000)
+        {
+            freq >>>= 1;
+            elTime >>>= 1;
+        }
+        if (elTime == 0)
+        {
+            elTime = 1;
+        }
+        return value * freq / elTime;
+    }
+    
+    static long GetCompressRating(int dictionarySize, long elapsedTime,
+            long size)
+    {
+        final long t = GetLogSize(dictionarySize) - (18 << kSubBits);
+        final long numCommandsForOne = 1060 + (t * t * 10 >> 2 * kSubBits);
+        final long numCommands = size * numCommandsForOne;
+        return MyMultDiv64(numCommands, elapsedTime);
+    }
+    
+    static long GetDecompressRating(long elapsedTime, long outSize, long inSize)
+    {
+        final long numCommands = inSize * 220 + outSize * 20;
+        return MyMultDiv64(numCommands, elapsedTime);
+    }
+    
+    static long GetTotalRating(int dictionarySize, long elapsedTimeEn,
+            long sizeEn, long elapsedTimeDe, long inSizeDe, long outSizeDe)
+    {
+        return (GetCompressRating(dictionarySize, elapsedTimeEn, sizeEn) + GetDecompressRating(
+                elapsedTimeDe, inSizeDe, outSizeDe)) / 2;
+    }
+    
+    static void PrintValue(long v)
+    {
+        String s = "";
+        s += v;
+        for (int i = 0; i + s.length() < 6; i++)
+        {
+            System.out.print(" ");
+        }
+        System.out.print(s);
+    }
+    
+    static void PrintRating(long rating)
+    {
+        PrintValue(rating / 1000000);
+        System.out.print(" MIPS");
+    }
+    
+    static void PrintResults(int dictionarySize, long elapsedTime, long size,
+            boolean decompressMode, long secondSize)
+    {
+        final long speed = MyMultDiv64(size, elapsedTime);
+        PrintValue(speed / 1024);
+        System.out.print(" KB/s  ");
+        long rating;
+        if (decompressMode)
+        {
+            rating = GetDecompressRating(elapsedTime, size, secondSize);
+        }
+        else
+        {
+            rating = GetCompressRating(dictionarySize, elapsedTime, size);
+        }
+        PrintRating(rating);
+    }
+    
+    static public int LzmaBenchmark(int numIterations, int dictionarySize)
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

This method contains 65 lines of code, not including blank lines or lines with only comments, Java punctuation characters, identifiers, or literals. By comparison, 99% of the methods in the CodeGuru reference dataset contain fewer lines of code. Large methods might be difficult to read and have logic that is hard to understand and test. We recommend that you simplify this method or break it into multiple methods.